packages/core/admin/server/services/permission/permissions-manager/query-builers.js

```diff
@@ -3,79 +3,69 @@
 // TODO: migration
 const _ = require('lodash');
 const { rulesToQuery } = require('@casl/ability/extra');
-const { VALID_REST_OPERATORS } = require('@strapi/utils');
 
-const ops = {
-  common: VALID_REST_OPERATORS.map(op => `$${op}`),
-  boolean: ['$or', '$and'],
-  cleanable: ['$elemMatch'],
+const operatorsMap = {
+  $in: '$in',
+  $nin: '$notIn',
+  $exists: '$notNull',
+  $gte: '$gte',
+  $gt: '$gt',
+  $lte: '$lte',
+  $lt: '$lt',
+  $eq: '$eq',
+  $ne: '$ne',
+  $and: '$and',
+  $or: '$or',
+  $not: '$not',
+};
+
+const mapKey = key => {
+  if (_.isString(key) && key.startsWith('$') && key in operatorsMap) {
+    return operatorsMap[key];
+  }
+  return key;
 };
 
 const buildCaslQuery = (ability, action, model) => {
   const query = rulesToQuery(ability, action, model, o => o.conditions);
-  return _.get(query, '$or[0].$and', {});
+  return _.get(query, '$or[0]', {});
```

Comment:
now that we support $or can we return the query directly ?

Reply:
I guess so, lets try